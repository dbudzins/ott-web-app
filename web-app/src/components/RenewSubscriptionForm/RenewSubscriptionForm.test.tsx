import React from 'react';
import { render } from '@testing-library/react';

import customer from '../../fixtures/customer.json';
import subscription from '../../fixtures/subscription.json';
import type { Subscription } from '../../../types/subscription';
import type { Customer } from '../../../types/account';

import RenewSubscriptionForm from './RenewSubscriptionForm';

describe('<RenewSubscriptionForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <RenewSubscriptionForm
        customer={customer as Customer}
        subscription={subscription as Subscription}
        onConfirm={vi.fn()}
        onClose={vi.fn()}
        error={null}
        submitting={false}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
