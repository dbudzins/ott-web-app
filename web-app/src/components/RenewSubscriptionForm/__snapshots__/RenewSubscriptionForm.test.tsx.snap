// Vitest Snapshot v1

exports[`<RenewSubscriptionForm> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="title"
    >
      renew_subscription.renew_your_subscription
    </h2>
    <p>
      renew_subscription.explanation
    </p>
    <div
      class="infoBox"
    >
      <p>
        <strong>
          Annual subscription (recurring)
        </strong>
         
        <br />
        renew_subscription.next_billing_date_on
      </p>
      <p
        class="price"
      >
        <strong>
          € 22,15
        </strong>
        <small>
          /
          periods.year
        </small>
      </p>
    </div>
    <button
      aria-disabled="false"
      class="button confirmButton primary fullWidth"
    >
      <span>
        renew_subscription.renew_subscription
      </span>
    </button>
    <button
      class="button default outlined fullWidth"
    >
      <span>
        renew_subscription.no_thanks
      </span>
    </button>
  </div>
</div>
`;
